api/src/adapters/sam_gov/client.py

```diff
@@ -0,0 +1,242 @@
+                    f.write(chunk)
+
+            # Build response object
+            extract_response = SamExtractResponse(
+                file_name=filename or os.path.basename(output_path),
+                file_size=content_length or os.path.getsize(output_path),
+                content_type=content_type,
+                sensitivity=SensitivityLevel.PUBLIC,
+                download_date=datetime.now(),
+            )
+
+            return extract_response
+
+        except requests.RequestException as e:
+            raise Exception(f"Request failed: {str(e)}") from e
+        except IOError as e:
+            raise IOError(f"Failed to save file: {str(e)}") from e
+        except Exception as e:
+            raise Exception(f"Error downloading extract: {str(e)}") from e
+
+    def _request(
+        self,
+        method: str,
+        endpoint: str,
+        params: dict[str, Any] | None = None,
+        data: dict[str, Any] | None = None,
+        stream: bool = False,
+        accept_content_type: str = "application/json",
+        **kwargs: Any,
+    ) -> requests.Response:
+        """Make a request to the SAM.gov API.
+
+        Args:
+            method: HTTP method to use.
+            endpoint: API endpoint to call.
+            params: Optional query parameters.
+            data: Optional request body data.
+            stream: Whether to stream the response.
+            accept_content_type: Content type to accept in the response.
+            **kwargs: Additional arguments to pass to requests.
+
+        Returns:
+            The HTTP response.
+
+        Raises:
+            requests.exceptions.RequestException: For various request errors.
+        """
+        # Validate that API key is provided
+        if not self.api_key:
+            raise ValueError("API key must be provided for SAM.gov API access")
+
+        # Validate that API URL is provided
+        if not self.api_url:
+            raise ValueError("API URL must be provided for SAM.gov API access")
+
+        # Use urljoin to handle URL path concatenation properly
+        url = urljoin(self.api_url, endpoint.lstrip("/"))
+
+        # Add API key to query parameters
+        if params is None:
+            params = {}
+        params["api_key"] = self.api_key
+
+        if "timeout" not in kwargs:
+            kwargs["timeout"] = 30  # Default timeout
+
+        # Get headers
+        headers = self._build_headers()
+
+        # Update Accept header if provided
+        headers["Accept"] = accept_content_type
+
+        try:
+            response = requests.request(
+                method, url, params=params, json=data, stream=stream, headers=headers, **kwargs
+            )
+
+            # Raise an exception for HTTP errors
+            response.raise_for_status()
+            return response
+
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Error when calling SAM.gov API: {e}")
+            raise
```

Comment:
logger.exception attaches the exception in a format that works with New Relic.
```suggestion
        except requests.exceptions.RequestException:
            logger.exception("Error when calling SAM.gov API")
            raise
```